Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -221,9 +221,29 @@ class PurchasesOrchestrator {
         }
     }
 
+    @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
+    func purchase(sk1Product: SK1Product,
+                  storeProductDiscount: StoreProductDiscount,
+                  package: Package?,
+                  completion: @escaping PurchaseCompletedBlock) {
+        self.paymentDiscount(forProductDiscount: storeProductDiscount,
+                             product: sk1Product) { [unowned self] discount, error in
+            guard let discount = discount else {
+                completion(nil, nil, error, false)
+                return
+            }
+
+            let payment = self.storeKitWrapper.payment(withProduct: sk1Product, discount: discount)
+            self.purchase(sk1Product: sk1Product,
+                          payment: payment,
+                          package: package,
+                          completion: completion)
+        }
+    }
+
     func purchase(sk1Product: SK1Product,
                   payment: SKMutablePayment,
-                  presentedOfferingIdentifier maybePresentedOfferingIdentifier: String?,
+                  package: Package?,
```

Comment:
Per @taquitos recommendation, this does seem like an overall safer approach to pass a `Package` if that is actually what this identifier was for 💡

Reply:
🤔 
If you have the package already, perhaps it'd be better to call `purchase(package:` instead? 
@vegaro I'm not super clear on which use case this version of the method covers, do you recall the intent behind `presentedOfferingIdentifier`?

Reply:
Was this method with `presentedOfferingIdentifier` ever public in an older version of the SDK? Seems like maybe it was carried over from that into orchestrator? I'm not actually sure so just throwing things out here 😊

Reply:
we talked about this in a sync-up today: 
- this version of the method was never public, we can do whatever's most useful to us
- looks like we should be able to just have the `package` without the `product` and `presentedOfferingIdentifier`, and get those from the `package`

Reply:
Looks like @NachoSoto took care of this #1160